src/redisearch_rs/c_entrypoint/types_ffi/src/lib.rs

```diff
@@ -462,77 +510,41 @@ pub unsafe extern "C" fn AggregateResult_AddChild(
     }
 }
 
-/// Create an iterator over the aggregate result. This iterator should be freed
-/// using [`AggregateResultIter_Free`].
+/// Get a view of the records stored inside the aggregate result.
 ///
 /// # Safety
 /// The following invariants must be upheld when calling this function:
 /// - `agg` must point to a valid `RSAggregateResult` and cannot be NULL.
 #[unsafe(no_mangle)]
-pub unsafe extern "C" fn AggregateResult_Iter(
+pub unsafe extern "C" fn AggregateResult_GetRecordsSlice(
     agg: *const RSAggregateResult<'static>,
-) -> *mut RSAggregateResultIter<'static> {
+) -> AggregateRecordsSlice {
     debug_assert!(!agg.is_null(), "agg must not be null");
 
     // SAFETY: Caller is to ensure that the pointer `agg` is a valid, non-null pointer to
     // an `RSAggregateResult`.
     let agg = unsafe { &*agg };
-    let iter = agg.iter();
```

Comment:
Do we still need the iterator implemenation on the aggregate after this?

Reply:
Even though we don't currently use it, I assume we'll switch over to it once the callers of this API move to Rust.